hooks/tk-multi-publish2/basic/nuke_publish_reviewsubmission.py

```diff
@@ -0,0 +1,333 @@
+            )
+        return {
+            "accepted": accepted,
+            "checked": True
+        }
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        review_submission_app = self.parent.engine.apps.get("tk-multi-reviewsubmission")
+        if review_submission_app is None:
+            error_msg = "Item cannot be validated. Review Submission App is not available."
+            self.logger.error(error_msg)
+            raise Exception(error_msg)
+        write_node_app = self.parent.engine.apps.get("tk-nuke-writenode")
+        if write_node_app is None:
+            error_msg = "Item cannot be validated. Write Node App is not available."
+            self.logger.error(error_msg)
+            raise Exception(error_msg)
+
+        # the render task will always render full-res frames when publishing. If we're
+        # in proxy mode in Nuke, that task will fail since there will be no full-res 
+        # frames rendered. The exceptions are if there is no proxy_render_template set 
+        # in the tk-nuke-writenode app, then the write node app falls back on the
+        # full-res template. Or if they rendered in full res and then switched to 
+        # proxy mode later. In this case, this is likely user error, so we catch it.
+        root_node = nuke.root()
+        proxy_mode_on = root_node['proxy'].value()
+        if proxy_mode_on:
+            error_msg = "You cannot publish to Screening Room while Nuke is in proxy " +\
+                        "mode. Please toggle proxy mode OFF and try again."
+            self.logger.error(error_msg)
+            raise Exception(error_msg)
+
+        publisher = self.parent
+
+        # populate the publish template on the item if found
+        publish_template_setting = settings.get("Publish Template")
+        publish_template = publisher.engine.get_template_by_name(
+            publish_template_setting.value)
+        if publish_template:
+            item.properties["publish_template"] = publish_template
+
+        return True
+
+    def publish(self, settings, item):
+        """
+        Executes the publish logic for the given item and settings.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        review_submission_app = self.parent.engine.apps.get("tk-multi-reviewsubmission")
+        if review_submission_app is None:
```

Comment:
We know the app is valid because of the check in `accept()`. I think we can take this out.

Reply:
removed